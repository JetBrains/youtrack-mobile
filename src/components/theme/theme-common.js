/* @flow */

import type {UIThemeColorsPaletteColors} from '../../flow/Theme';

export const themeColorsPalette: UIThemeColorsPaletteColors = {
  $greyBackground: 'rgba(184, 209, 229, 0.15)',
  $greyColor: '#b8d1e5',

  $redBackground: 'rgba(219, 88, 96, 0.15)',
  $redColor: '#db5860',

  $yellowBackground: 'rgba(237, 162, 0, 0.15)',
  $yellowColor: '#eda200',

  $greenBackground: 'rgba(89, 168, 105, 0.15)',
  $greenColor: '#59a869',

  $blueBackground: 'rgba(0, 142, 255, 0.15)',
  $blueColor: '#008eff',
};
