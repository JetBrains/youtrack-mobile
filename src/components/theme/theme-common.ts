import type {UIThemeColorsPaletteColors} from 'types/Theme';
export const themeColorsPalette: UIThemeColorsPaletteColors = {
  $greyBackground: 'rgba(184, 209, 229, 0.15)',
  $greyColor: 'rgba(59, 107, 137, 0.35)',
  $redBackground: 'rgba(219, 88, 96, 0.15)',
  $redColor: '#db5860',
  $yellowBackground: 'rgba(237, 162, 0, 0.15)',
  $yellowColor: '#eda200',
  $greenBackground: 'rgba(89, 168, 105, 0.15)',
  $greenColor: '#59a869',
  $blueBackground: 'rgba(0, 142, 255, 0.15)',
  $blueColor: '#008eff',
};
