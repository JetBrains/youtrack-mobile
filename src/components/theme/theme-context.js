/* @flow */

import React from 'react';

export const ThemeContext = React.createContext<any>(null);
