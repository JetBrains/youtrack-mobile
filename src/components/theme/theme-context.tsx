import React from 'react';
export const ThemeContext: React.Context<any> = React.createContext<any>(null);
