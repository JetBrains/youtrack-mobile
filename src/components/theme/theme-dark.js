/* @flow */

import type {UITheme} from '../../flow/Theme';

const dark: UITheme = {
  dark: true,
  mode: 'ytdark',
  name: 'dark',
  barStyle: 'light-content',
  androidSummaryFontWeight: '400',

  colors: {
    $background: '#1c1c1e',
    $boxBackground: '#2c2c2e',

    $error: '#ee0e0e',

    $text: '#e1e1e1',
    $textSecondary: '#b0b0b0',
    $textButton: '#FFFFFF',

    $link: '#FF008C',
    $linkLight: 'rgba(254, 0, 130, 0.3)',

    $disabled: '#5d5d5d',

    $icon: '#9f9f9f',
    $iconAccent: '#b8d1e5',

    $mask: '#FFFFFF57',
    $navigation: '#696969',

    $separator: 'rgba(210, 254, 254, 0.15)'
  }
};

export default dark;
