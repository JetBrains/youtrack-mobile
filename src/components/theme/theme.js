/* @flow */

import {Appearance} from 'react-native-appearance';
import EStyleSheet from 'react-native-extended-stylesheet';

import lightTheme from './theme-light';
import darkTheme from './theme-dark';

import type {UITheme} from '../../flow/Theme';

export const DEFAULT_THEME: UITheme = lightTheme;
export const DEFAULT_MODE: string = lightTheme.name;

export const getSystemMode = (): string => Appearance.getColorScheme() || DEFAULT_MODE;
export const DEFAULT_SYSTEM_MODE: string = getSystemMode();

export const getUITheme = (themeName: string): UITheme => {
  if ([lightTheme.name, darkTheme.name].includes(themeName)) {
    return themeName === DEFAULT_MODE ? lightTheme : darkTheme;
  }
  return DEFAULT_THEME;
};

export const buildStyles = (mode: string = DEFAULT_SYSTEM_MODE, uiTheme: UITheme = DEFAULT_THEME) => {
  EStyleSheet.build({
    $theme: mode,
    $resolved: uiTheme.colors.$icon,
    $shadowColor: uiTheme.colors.$icon,
    $link: uiTheme.colors.$link,
    $iconAccent: uiTheme.colors.$iconAccent,
    $background: uiTheme.colors.$background,
    ...uiTheme.colors
  });
};

