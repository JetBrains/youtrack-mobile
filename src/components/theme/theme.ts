import {Appearance} from 'react-native';
import EStyleSheet from 'react-native-extended-stylesheet';
import AsyncStorage from '@react-native-async-storage/async-storage';
import darkTheme from './theme-dark';
import lightTheme from './theme-light';
import {THEME_MODE_KEY} from '../storage/storage';
import type {UITheme} from 'types/Theme';
export const DEFAULT_THEME: UITheme = lightTheme;
export const getSystemThemeMode = (): any => Appearance.getColorScheme();
export const themes: UITheme[] = [lightTheme, darkTheme];
export const getUITheme = (mode: string): UITheme => {
  const theme: UITheme | null | undefined = themes.reduce(
    (theme: UITheme | null | undefined, it: UITheme) =>
      it.mode.includes(mode) ? it : theme,
    null,
  );
  return theme || DEFAULT_THEME;
};
export const buildStyles = (mode: string, uiTheme: UITheme) => {
  EStyleSheet.build({
    $theme: mode,
    $androidSummaryFontWeight: uiTheme.androidSummaryFontWeight,
    $link: uiTheme.colors.$link,
    $resolved: uiTheme.colors.$icon,
    $disabled: uiTheme.colors.$disabled,
    $icon: uiTheme.colors.$icon,
    ...uiTheme.colors,
  });
};
export const getThemeMode = async (): Promise<string> => {
  let mode: string;

  try {
    const storedMode: string | null | undefined = JSON.parse(
      await AsyncStorage.getItem(THEME_MODE_KEY),
    );
    mode = storedMode || getSystemThemeMode();
  } catch (e) {
    mode = DEFAULT_THEME.mode;
  }

  return mode;
};
