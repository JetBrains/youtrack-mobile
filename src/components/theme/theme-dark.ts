/* @flow */

import {themeColorsPalette} from './theme-common';

import type {UITheme} from 'flow/Theme';

const dark: UITheme = {
  dark: true,
  mode: 'ytdark',
  name: 'dark',
  barStyle: 'light-content',
  androidSummaryFontWeight: '400',

  colors: {
    $background: '#1c1c1e',
    $boxBackground: '#2c2c2e',
    $dimBackground: 'rgba(0,0,0,0.7)',

    $error: '#C22731',

    $text: '#e1e1e1',
    $textSecondary: '#b0b0b0',
    $textButton: '#FFFFFF',

    $link: '#FF008C',
    $linkLight: 'rgba(254, 0, 130, 0.3)',

    $disabled: '#5d5d5d',

    $icon: '#9f9f9f',
    $iconAccent: '#577996',

    $mask: '#FFFFFF57',
    $navigation: '#696969',

    $separator: 'rgba(210, 254, 254, 0.15)',

    ...themeColorsPalette,
  },
};

export default dark;
