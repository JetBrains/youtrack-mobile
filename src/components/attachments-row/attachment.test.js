import React from 'react';

import {shallow} from 'enzyme';

import Attachment from './attachment';
import {DEFAULT_THEME} from '../theme/theme';

describe('<Attachment/>', () => {

  let wrapper;
  let instance;
  let attachmentMock;
  const attachmentMockName = 'attachName';
  const attachmentMockUrl = 'http://example.com?attach';

  beforeEach(() => {
    attachmentMock = {
      id: 'attachId',
      name: attachmentMockName,
      url: attachmentMockUrl
    };
  });


  describe('Render image', () => {
    it('should render image', () => {
      renderImage();

      expect(findByTestId('attachmentImage')).toHaveLength(1);
      expect(findByTestId('attachmentSvg')).toHaveLength(0);
      expect(findByTestId('attachmentFile')).toHaveLength(0);
      expect(findByTestId('attachmentRemove')).toHaveLength(0);
    });

    it('should render remove image button', () => {
      renderImage(true);
      expect(findByTestId('attachmentRemove')).toHaveLength(1);
    });
  });


  describe('Render SVG', () => {
    it('should render SVG', () => {
      renderWithMimeType('image/svg+xml');

      expect(findByTestId('attachmentImage')).toHaveLength(0);
      expect(findByTestId('attachmentSvg')).toHaveLength(1);
      expect(findByTestId('attachmentFile')).toHaveLength(0);
    });
  });


  describe('Render file', () => {
    it('should render file', () => {
      renderFile();

      expect(findByTestId('attachmentImage')).toHaveLength(0);
      expect(findByTestId('attachmentSvg')).toHaveLength(0);
      expect(findByTestId('attachmentFile')).toHaveLength(1);
    });
  });


  describe('showImageAttachment', () => {
    it('should show image attachment', async () => {
      renderImage();
      instance.showImageAttachment = jest.fn();
      instance.forceUpdate();

      findByTestId('attachment').simulate('press');

      expect(instance.showImageAttachment).toHaveBeenCalledWith(Object.assign(
        {},
        attachmentMock,
        {mimeType: 'image/png'},
        {url: attachmentMockUrl}
      ));
    });
  });


  describe('openAttachmentUrl', () => {
    it('should show image attachment', async () => {
      renderFile();
      instance.openAttachmentUrl = jest.fn();
      instance.forceUpdate();

      findByTestId('attachment').simulate('press');
      expect(instance.openAttachmentUrl).toHaveBeenCalledWith(attachmentMockName, attachmentMockUrl);
    });
  });


  function renderImage(canRemoveImage) {
    renderWithMimeType('image/png', canRemoveImage);
  }

  function renderFile(canRemoveImage) {
    renderWithMimeType('application/stream', canRemoveImage);
  }

  function renderWithMimeType(mimeType: string, canRemoveImage: boolean) {
    render({
      mimeType: mimeType
    }, canRemoveImage);
  }

  function render(attachment: Attachment, canRemoveImage: boolean) {
    wrapper = doShallow(Object.assign({}, attachmentMock, attachment), canRemoveImage);
    instance = wrapper.instance();
  }

  function findByTestId(testId) {
    return wrapper && wrapper.find({testID: testId});
  }

  function doShallow(attachment: Attachment, canRemoveImage: boolean = false) {
    return shallow(
      <Attachment
        attach={attachment}
        canRemoveImage={canRemoveImage}
        uiTheme={DEFAULT_THEME}
      />
    );
  }
});
