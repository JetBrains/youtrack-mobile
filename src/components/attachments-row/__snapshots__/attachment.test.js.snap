// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Attachment/> Render should match a snapshot 1`] = `
<View
  key="attachId"
  style={null}
>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    testID="attachment"
  >
    <withAnimatable(View)
      animation={null}
      delay={0}
      direction="normal"
      duration={700}
      easing="ease-out-quart"
      iterationCount={1}
      onAnimationBegin={[Function]}
      onAnimationEnd={[Function]}
      testID="attachmentImage"
      useNativeDriver={true}
    >
      <ImageProgress
        errorContainerStyle={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        indicatorContainerStyle={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        onError={[Function]}
        renderIndicator={[Function]}
        source={
          Object {
            "headers": null,
            "uri": undefined,
          }
        }
        style={
          Object {
            "borderColor": "rgba(0, 0, 0, 0.15)",
            "borderRadius": 3,
            "borderWidth": 1,
            "height": 64,
            "marginRight": 8,
            "overflow": "hidden",
            "width": 120,
          }
        }
        threshold={50}
      />
    </withAnimatable(View)>
  </TouchableOpacity>
</View>
`;
